lib/puppet/indirector/report/json.rb

```diff
@@ -0,0 +1,34 @@
+require 'puppet/transaction/report'
+require 'puppet/indirector/json'
+
+class Puppet::Transaction::Report::Json < Puppet::Indirector::JSON
+  include Puppet::Util::SymbolicFileMode
```

Comment:
The yaml report terminus does support symbolic file modes, but I'm thinking we shouldn't here either. It would simplify this implementation as we'd just need to overrride `path` below.